modules/image_processors/LineBoundingBoxDetector.h

```diff
@@ -0,0 +1,69 @@
+/**
+ * @file   LineBoundingBoxDetector.h
+ * @brief  ライントレース用の画像処理クラス
+ * @author takuchi17 miyahara046 HaruArima08
+ */
+
+#ifndef LINE_BOUNDING_BOX_DETECTOR_H
+#define LINE_BOUNDING_BOX_DETECTOR_H
+
+#include "BoundingBoxDetector.h"
+
+// 解像度の最小・最大値
+struct ResolutionRange {
+  static constexpr int MIN_WIDTH = 320;
+  static constexpr int MIN_HEIGHT = 240;
+  static constexpr int MAX_WIDTH = 1920;
+  static constexpr int MAX_HEIGHT = 1080;
+};
+
+class LineBoundingBoxDetector : public BoundingBoxDetector {
+ public:
+  /**
+   * @brief デフォルトのROIおよび解像度を使用するコンストラクタ
+   * @param lowerHSV ライントレース対象の色の下限HSV値
+   * @param upperHSV ライントレース対象の色の上限HSV値
+   */
+  LineBoundingBoxDetector(const cv::Scalar& _lowerHSV, const cv::Scalar& _upperHSV);
+
+  /**
+   * @brief ROIを指定するオーバーロードコンストラクタ
+   * @param lowerHSV ライントレース対象の色の下限HSV値
+   * @param upperHSV ライントレース対象の色の上限HSV値
+   * @param roi 注目領域
+   */
+  LineBoundingBoxDetector(const cv::Scalar& lowerHSV, const cv::Scalar& upperHSV,
+                          const cv::Rect& roi);
+
+  /**
+   * @brief ROIと解像度を指定するオーバーロードコンストラクタ
+   * @param lowerHSV ライントレース対象の色の下限HSV値
+   * @param upperHSV ライントレース対象の色の上限HSV値
+   * @param roi 注目領域
+   * @param resolution 画像処理に用いる解像度
+   */
+  LineBoundingBoxDetector(const cv::Scalar& lowerHSV, const cv::Scalar& upperHSV,
+                          const cv::Rect& roi, const cv::Size& resolution);
```

Comment:
cpp ファイルでの定義時と、引数の変数名が異なっているから、合わせたほうが良いかも

```suggestion
  LineBoundingBoxDetector(const cv::Scalar& _lowerHSV, const cv::Scalar& _upperHSV,
                          const cv::Rect& _roi, const cv::Size& _resolution);
```

さらに言うと、オーバーロードしているコンストラクタごとに下ハイフンがついたり消えてたりするから、統一したほうがいいと思うー